src/allocator.rs

```diff
@@ -152,6 +153,16 @@ impl Allocator {
         node_from_number(self, &v)
     }
 
+    pub fn new_g1(&mut self, g1: G1Projective) -> Result<NodePtr, EvalErr> {
+        let g1: G1Affine = g1.into();
+        self.new_atom(&g1.to_compressed())
```

Comment:
Ideally, we do this lazily. It's possible no one ever asks for the bytes representation of this object.

Reply:
yes. My plan is to first raise the abstraction level of the Allocator (which this patch is part of). Then I update the Allocator to preserve the native representations of these types in memory.

This function is a temporary step along that path.

Reply:
Okay, so for now, I suppose using `G1Affine` is reasonable since all you're really doing is generating the atom blob. Once we cache the internal representations, we definitely should cache them as `G1Projective`. We only ever should convert to `G1Affine` if someone wants the atom blob representation.

You might even consider also consider caching `G1Affine` representations, as the cheapest op form is to take affine coordinates as inputs and generate projective coordinates as output (since z=1 always for affine, which is why it's elided). But this is a hyper-optimization and I'd say not really necessary.

Reply:
I don't believe this PR is making anything worse than it already is btw. The intention is to move the conversions into the Allocator, to give the allocator control over whether and when conversions are made ,in a later patch. That later patch will be quite large and complex, so I'm trying to front-load as many changes as possible that are somewhat independent